pkg/deploy/lattice/target_group_manager.go

```diff
@@ -288,44 +289,43 @@ func (s *defaultTargetGroupManager) findTargetGroup(
 	if len(arns) == 0 {
 		return nil, nil
 	}
-	// Tag fields guarantee one result, as there can be only one target group for one service/route combination.
-	// We move forward but log this situation to help troubleshooting
-	if len(arns) > 1 {
-		s.log.Warnw("Target groups with conflicting tags found", "arns", arns)
-	}
-	arn := arns[0]
 
-	latticeTg, err := s.cloud.Lattice().GetTargetGroupWithContext(ctx, &vpclattice.GetTargetGroupInput{
-		TargetGroupIdentifier: &arn,
-	})
-	if err != nil {
-		return nil, services.IgnoreNotFound(err)
-	}
+	for _, arn := range arns {
+		latticeTg, err := s.cloud.Lattice().GetTargetGroupWithContext(ctx, &vpclattice.GetTargetGroupInput{
+			TargetGroupIdentifier: &arn,
+		})
+		if err != nil {
+			if services.IsNotFoundError(err) {
+				continue
+			}
+			return nil, err
+		}
 
-	// we ignore create failed status, so may as well check for it first
-	status := aws.StringValue(latticeTg.Status)
-	if status == vpclattice.TargetGroupStatusCreateFailed {
-		return nil, nil
-	}
+		// we ignore create failed status, so may as well check for it first
+		status := aws.StringValue(latticeTg.Status)
+		if status == vpclattice.TargetGroupStatusCreateFailed {
+			continue
+		}
 
-	// Double-check the immutable fields to ensure TG is valid
-	match, err := s.IsTargetGroupMatch(ctx, modelTargetGroup, &vpclattice.TargetGroupSummary{
-		Arn:           latticeTg.Arn,
-		Port:          latticeTg.Config.Port,
-		Protocol:      latticeTg.Config.Protocol,
-		IpAddressType: latticeTg.Config.IpAddressType,
-		Type:          latticeTg.Type,
-		VpcIdentifier: latticeTg.Config.VpcIdentifier,
-	}, nil) // we already know that tags match
-	if err != nil {
-		return nil, err
-	}
-	if match {
-		switch status {
-		case vpclattice.TargetGroupStatusCreateInProgress, vpclattice.TargetGroupStatusDeleteInProgress:
-			return nil, errors.New(LATTICE_RETRY)
-		case vpclattice.TargetGroupStatusDeleteFailed, vpclattice.TargetGroupStatusActive:
-			return latticeTg, nil
+		// Check the immutable fields to ensure TG is valid
+		match, err := s.IsTargetGroupMatch(ctx, modelTargetGroup, &vpclattice.TargetGroupSummary{
+			Arn:           latticeTg.Arn,
+			Port:          latticeTg.Config.Port,
+			Protocol:      latticeTg.Config.Protocol,
+			IpAddressType: latticeTg.Config.IpAddressType,
+			Type:          latticeTg.Type,
+			VpcIdentifier: latticeTg.Config.VpcIdentifier,
+		}, nil) // we already know that tags match
```

Comment:
Just wondering should we add a `Protocol`(or other tg fields) tags for TargetGroup?
So that to make sure one `FindResourcesByTags()` api could definitely exactly match one TG?

 (not in this PR)

Reply:
Not sure if it is necessary. Actually.. I wanted to remove as many tags as possible. I "had to" add protocolVersion in the tag because it is not available in TGSummary, so it needed extra Get() call to be figured out. Ideally VPC Lattice should just return protocolVersion in the summary instead.